### PR TITLE
Settings/Open Source Licenses

### LINKED ISSUE #104: Settings/Open Source Licenses
**Description:** As a user, I want to be able to view the open source licenses of the libraries used in the app. This will allow me to have transparency about the open source software components and their licenses.

**Acceptance Criteria:**

1. When I navigate to the settings page in the mobile app, I should see a clearly labeled "Open Source Licenses" button.
2. When I tap on the "Open Source Licenses" button, it should open a new screen or dialog displaying the list of open source licenses used in the app.
3. The list should include the names of the libraries, their licenses, and any additional information required by the licenses.
4. The user should be able to scroll and read through the list of licenses.
5. There should be a close button or back navigation option to return to the settings page after viewing the licenses.

**Additional Information:** You may consider using the [Google Play Services OSS Licenses Plugin](https://github.com/google/play-services-plugins/tree/master/oss-licenses-plugin) as a possible way of implementing the functionality to generate and display the open source licenses in the app.

### PR DESCRIPTION
**Pull request by KatyaSmishlyaeva**
## Summary

Added Open Source Licenses to Settings

## Reasons

User will be able to view the open source licenses of the libraries used in the app. This will allow user to have transparent information about the open source software components and their licenses.

## References

closes #104

### CODE REVIEW

File: app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    val licensesList: StateFlow<List<LibraryItem>> = _licensesList.asStateFlow()
+
+    init {
+        loadLicenses(application)
+    }
+
+    private fun loadLicenses(context: Context) {
+        viewModelScope.launch(appDispatchers.io) {
+            val licensesData = context.resources
+                .openRawResource(R.raw.third_party_licenses)
```
Comment (EvgeniyFilippov) [inline app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt:34]: need to add handling of possible exceptions (NotFoundException)

File: app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                .readBytes()
+            val licensesMetadataReader = context.resources
+                .openRawResource(R.raw.third_party_license_metadata)
+                .bufferedReader()
+
+            _licensesList.value =
+                licensesMetadataReader.use { reader -> reader.readLines() }.map { line ->
+                    val (section, name) = line.split(" ", limit = 2)
+                    val (startOffset, length) = section.split(":", limit = 2).map(String::toInt)
```
Comment (EvgeniyFilippov) [inline app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt:43]: please, move "2" to const

File: app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    val licenseData =
+                        licensesData.sliceArray(startOffset until startOffset + length)
+                    val licenseText = licenseData.toString(Charset.forName("UTF-8"))
```
Comment (EvgeniyFilippov) [inline app/src/main/kotlin/com/softteco/template/ui/feature/openSourceLicenses/OpenSourceLicensesViewModel.kt:46]: can use Charsets.UTF_8 instead Charset.forName("UTF-8")

**Review by EvgeniyFilippov — APPROVED** [review]

**Comment by bstolyarov** [comment]
How does it look? On the debug version, I see only this alert: 

Screenshot_20231212_105929

**Comment by KatyaSmishlyaeva** [comment]
On the release version, it looks like this
photo_1
photo_2
photo_3

**Review by bstolyarov — APPROVED** [review]

**Pull request merged**